daemon/api.go

```diff
@@ -563,7 +563,7 @@ func (inst *snapInstruction) install() (*state.Change, error) {
 		flags = 0
 	}
 	msg := fmt.Sprintf(i18n.G("Install %q snap"), inst.pkg)
-	if inst.Channel != "stable" {
+	if inst.Channel != "stable" && inst.Channel != "" {
```

Comment:
Can you please pick the test from #1012?  It's just for install, but there's likely an equivalent test there that can be updated for refresh.